### PR TITLE
Create Channel UI to specs

### PR DESCRIPTION
**Pull request by Dorvaryn**
This PR implements design specs for the Create Channel screen

| Before | After |
| --- | --- | 
beforehint|afterhint
beforetype|aftertyped

### CODE REVIEW

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java [hunk 1]
```diff
@@ -32,22 +33,27 @@ protected void onFinishInflate() {
         View.inflate(getContext(), R.layout.merge_create_channel_view, this);
         newChannelName = Views.findById(this, R.id.newChannelName);
         privateChannelSwitch = Views.findById(this, R.id.privateChannelSwitch);
-        createButton = Views.findById(this, R.id.createButton);
+        toolbar = Views.findById(this, R.id.toolbar);
+        toolbar.setTitle(R.string.createChannel);
+        toolbar.inflateMenu(R.menu.new_channel_menu);
+        toolbar.setNavigationIcon(R.drawable.ic_clear_white_24dp);
```
Comment (takecare) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: move `setUpToolbar()`?

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java [hunk 2]
```diff
@@ -32,22 +33,27 @@ protected void onFinishInflate() {
...
 
     @Override
     public void attach(final ChannelCreationListener channelCreationListener) {
         this.channelCreationListener = channelCreationListener;
         newChannelName.addTextChangedListener(channelNameTextWatcher);
-        createButton.setOnClickListener(createButtonClickListener);
-        createButton.setEnabled(false);
+        toolbar.setOnMenuItemClickListener(menuItemClickListener);
+        toolbar.setNavigationOnClickListener(navigationOnClickListener);
+        toolbar.getMenu().getItem(0).setEnabled(false);
```
Comment (takecare) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: what's this? can we have a more descriptive name?

File: Firebase/android/app/src/main/res/drawable/ic_clear_white_24dp.xml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (takecare) [inline Firebase/android/app/src/main/res/drawable/ic_clear_white_24dp.xml]: 😮  👍

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/NewChannelActivity.java [hunk 4]
```diff
@@ -14,13 +14,12 @@
 public class NewChannelActivity extends BaseActivity {
 
     private NewChannelPresenter newChannelPresenter;
+    private Toolbar toolbar;
```
Comment (gbasile) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/NewChannelActivity.java]: [iOS-probably-dummy-question] It's not clear for me why you are using this toolbar here. Where do you use it?

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java [hunk 5]
```diff
@@ -99,17 +105,30 @@ private boolean isEmojiComponent(char c) {
 
         @Override
         public void afterTextChanged(Editable s) {
-            createButton.setEnabled(s.length() > 0 && isValidInput);
+            toolbar.getMenu().getItem(0).setEnabled(s.length() > 0 && isValidInput);
```
Comment (gbasile) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: Is it safe to get items by their position?

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java (hunk 1 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: 👍 Will extract

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java (hunk 2 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: 👍 I was planning on extracting this

File: Firebase/android/app/src/main/res/drawable/ic_clear_white_24dp.xml (hunk 3 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/res/drawable/ic_clear_white_24dp.xml]: This is fixed in the next PR let's leave it as it is on this one.

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/NewChannelActivity.java (hunk 4 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/NewChannelActivity.java]: 👍 I forgot to remove it after moving the toolbar out of this class.

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java [hunk 6]
```diff
@@ -99,17 +105,30 @@ private boolean isEmojiComponent(char c) {
...
                 setChannelNameError(R.string.only_single_emoji_allowed);
             }
         }
     };
 
-    private final OnClickListener createButtonClickListener = new OnClickListener() {
+    private final Toolbar.OnMenuItemClickListener menuItemClickListener = new Toolbar.OnMenuItemClickListener() {
+        @Override
+        public boolean onMenuItemClick(MenuItem item) {
+            switch (item.getItemId()) {
```
Comment (frapontillo) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: Do you plan on adding more stuff to the menu? if not, I'd go with a plain old `if`.

File: Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml [hunk 7]
```diff
@@ -3,25 +3,46 @@
...
     android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:text="@string/private_channel" />
+    android:layout_height="48dp"
+    android:background="@color/background_grey_alt"
+    android:paddingLeft="16dp"
+    android:paddingRight="16dp"
+    android:text="@string/private_channel"
+    android:textColor="@color/text_grey"
+    android:textSize="14sp" />
```
Comment (frapontillo) [inline Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml]: Maybe extract text sizes from this file?

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java (hunk 5 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: Damn I forgot to move this to a find by Id. 👍 My bad

File: Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml [hunk 8]
```diff
@@ -3,25 +3,46 @@
...
+  <include layout="@layout/toolbar_non_elelvated_layout" />
 
   <Switch
     android:id="@+id/privateChannelSwitch"
     android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:text="@string/private_channel" />
+    android:layout_height="48dp"
+    android:background="@color/background_grey_alt"
+    android:paddingLeft="16dp"
```
Comment (frapontillo) [inline Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml]: This padding looks pretty much as a default in here, you could extract it if it's used in other layouts too.

File: Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml (hunk 7 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml]: This is a first pass, I'll be extracting all styles when doing the review sitting down with @qqipp for design adjustments.

File: Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml (hunk 8 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/res/layout/merge_create_channel_view.xml]: This is a first pass, I'll be extracting all styles when doing the review sitting down with @qqipp for design adjustments.

File: Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java (hunk 6 above)
Comment (Dorvaryn) [inline Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java]: 👍

**Comment by ouchadam** [comment]
looking good ! :scroll: :

**Comment by Dorvaryn** [comment]
Can I get a merge ?

**Comment by frapontillo** [comment]
MEEEERGEEE

**Pull request merged**